model/patch/patch.go

```diff
@@ -973,3 +975,41 @@ func (p PatchesByCreateTime) Less(i, j int) bool {
 func (p PatchesByCreateTime) Swap(i, j int) {
 	p[i], p[j] = p[j], p[i]
 }
+
+func GetCollectiveStatus(allPatches []Patch) string {
```

Comment:
We shouldn't have the same function copy/pasted in different places. We should find a place that all the directories can access and keep the function there (I would've thought that was model/patch though). We should also always document public functions.

Reply:
one is for Patch and one is for APIPatch. I don't know if there is a clean way to combine them. 
Where do we document public functions?

Reply:
I can extract most of the logic though. I'll do that.

Reply:
never mind. I figured out a way to combine them. I'll ping you when it's ready for re-review.

Reply:
Done. Thank you for pointing it out. This setup was left over from how it was set up when I was sorting on the patch, but now that we are iterating it should have been combined.